spark/datadog_checks/spark/spark.py

```diff
@@ -215,79 +221,64 @@ def check(self, instance):
         if master_address is None:
-            master_address = instance.get(DEPRECATED_MASTER_ADDRESS)
+            master_address = self.instance.get(DEPRECATED_MASTER_ADDRESS)
 
             if master_address:
                 self.log.warning(
                     'The use of `%s` is deprecated. Please use `%s` instead.', DEPRECATED_MASTER_ADDRESS, MASTER_ADDRESS
                 )
             else:
-                raise Exception('URL for `%s` must be specified in the instance configuration' % MASTER_ADDRESS)
+                raise ConfigurationError(
+                    'URL for `%s` must be specified in the instance configuration' % MASTER_ADDRESS
+                )
 
         return master_address
 
-    def _get_request_url(self, instance, url):
+    def _get_request_url(self, url):
         """
         Get the request address, build with proxy if necessary
         """
         parsed = urlparse(url)
 
         _url = url
-        if not (parsed.netloc and parsed.scheme) and is_affirmative(instance.get('spark_proxy_enabled', False)):
-            master_address = self._get_master_address(instance)
-            _url = urljoin(master_address, parsed.path)
+        if not (parsed.netloc and parsed.scheme) and is_affirmative(self.instance.get('spark_proxy_enabled', False)):
+            _url = urljoin(self.master_address, parsed.path)
 
         self.log.debug('Request URL returned: %s', _url)
         return _url
 
-    def _get_running_apps(self, instance):
+    def _get_running_apps(self):
         """
         Determine what mode was specified
         """
-        tags = instance.get('tags', [])
-        if tags is None:
-            tags = []
-        master_address = self._get_master_address(instance)
-        # Get the cluster name from the instance configuration
-        cluster_name = instance.get('cluster_name')
-        if cluster_name is None:
-            raise Exception('The cluster_name must be specified in the instance configuration')
-        tags.append('cluster_name:%s' % cluster_name)
-        tags = list(set(tags))
-        # Determine the cluster mode
-        cluster_mode = instance.get(SPARK_CLUSTER_MODE)
-        if cluster_mode is None:
-            self.log.warning(
-                'The value for `spark_cluster_mode` was not set in the configuration. Defaulting to "%s"',
-                SPARK_YARN_MODE,
-            )
-            cluster_mode = SPARK_YARN_MODE
+        tags = list(self.tags)
+        tags.append('cluster_name:%s' % self.cluster_name)
 
-        if cluster_mode == SPARK_STANDALONE_MODE:
+        if self.cluster_mode == SPARK_STANDALONE_MODE:
             # check for PRE-20
-            pre20 = is_affirmative(instance.get(SPARK_PRE_20_MODE, False))
-            return self._standalone_init(master_address, pre20, tags)
+            pre20 = is_affirmative(self.instance.get(SPARK_PRE_20_MODE, False))
+            return self._standalone_init(self.master_address, pre20, tags)
```

Comment:
I think we don't need to continue using master_address as a param since we set it to `self.master_address`?